cypress/page/download.page.ts

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable require-jsdoc */
+class DownloadFilePage {
+  private downloadUrlPage: string;
+  private typeText: string;
+  private typePdf: string;
+  private generateTextFile: string;
+  private generatePdfFile: string;
+  private linkDownloadText: string;
+  private linkDownloadPdf: string;
+
+  constructor() {
+    this.downloadUrlPage = "demo.automationtesting.in/FileDownload.html";
+    this.typeText = ".form-group #textbox";
+    this.typePdf = ".form-group #pdfbox";
+    this.generateTextFile = "button#createTxt";
+    this.generatePdfFile = "button#createPdf";
+    this.linkDownloadText = "#link-to-download";
+    this.linkDownloadPdf = "#pdf-link-to-download";
+  }
+
+  public visitDownloadPage(): void {
+    cy.visit(this.downloadUrlPage);
+  }
+
+  public downloadTextData(textTest: string): void {
+    cy.get(this.typeText).type(textTest);
+    cy.get(this.generateTextFile).click();
+    cy.get(this.linkDownloadText).click();
+  }
```

Comment:
I will separate this method in two methods, as we are performing two different actions, first, enter the data in the text file section, and then, download the file.